pkg/netpol/eval/internal/k8s/policy_connections.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package k8s
+
+import (
+	"fmt"
+
+	apisv1a "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+
+	"github.com/np-guard/netpol-analyzer/pkg/internal/netpolerrors"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
+)
+
+// PolicyConnections : stores connections between two peers which is collected mainly from admin-network-policies and adjusted with
+// connections from network-policies/ baseline-admin-network-policies or system-default allowed connections as following:
+// - PASS connections will be enforced by allowed connections from network-policies, or if not captured by network-policies,
+// by connections from baseline-admin-network-policies or system-default connections.
+// - traffic that has no match in admin-network-policies will be determined also by network-policies or if not captured by network-policies,
+// by baseline-admin-network-policies or system default.
+type PolicyConnections struct {
+	// AllowedConns allowed connections-set between two peers
+	AllowedConns *common.ConnectionSet
+	// PassConns connections between two peers that was passed by admin-network-policy to policies with lower priority
```

Comment:
documentation on `pass` from `apis/v1alpha1/adminnetworkpolicy_types.go`:
```
Pass: instructs the selected traffic to skip any remaining ANP rules, and
 then pass execution to any NetworkPolicies that select the pod.
 If the pod is not selected by any NetworkPolicies then execution
 is passed to any BaselineAdminNetworkPolicies that select the pod.
```
Thus, using here the term " policies with lower priority" is confusing, as the term "priority" is relevant for ANP as a separate field.
I would change comment, to "delegate decision to next layer of policies, NetworkPolicies or BaselineAdminNetworkPolicies    resources"

Reply:
Is there a test that covers a case with "pass" rule after which there are other rules with different action, that demonstrates that the decision is delegated to next layer of policies? (thus making the next ANP rule after "pass" redundant)?

Reply:
> documentation on `pass` from `apis/v1alpha1/adminnetworkpolicy_types.go`:
> 
> ```
> Pass: instructs the selected traffic to skip any remaining ANP rules, and
>  then pass execution to any NetworkPolicies that select the pod.
>  If the pod is not selected by any NetworkPolicies then execution
>  is passed to any BaselineAdminNetworkPolicies that select the pod.
> ```
> 
> Thus, using here the term " policies with lower priority" is confusing, as the term "priority" is relevant for ANP as a separate field. I would change comment, to "delegate decision to next layer of policies, NetworkPolicies or BaselineAdminNetworkPolicies resources"

done

Reply:
> Is there a test that covers a case with "pass" rule after which there are other rules with different action, that demonstrates that the decision is delegated to next layer of policies? (thus making the next ANP rule after "pass" redundant)?

yes 
example:
`tests\anp_test_4_with_priority_chang_pass_to_banp` : has 2 ANPs with same connections;
one with `pass` the second with `deny` (the `pass` has precedence) so we see that the BANP determined the conn to be allowed.

and :
`tests\anp_test_4` is the same test; but the ANP with `deny` has the precedence; 

i have added a `diff` tests with both so we really see that the connections are denied in one and allowed in the other one.
(both are have connlist outputs too)

Reply:
thanks, will take a look.
Is there an example with both rules (pass and another one which is not pass ) in the same policy?

Reply:
yes (at least one), 
example:
`anp_test_7` : there is a "deny" rule from slytherin and a `pass` rule from slytherin : the result is deny since the "deny" rule is coming first
but in `anp_test_7_swapping_rules` : those rules were swapped, so the `pass` rule is coming first and the result is `allow` (passed to system-default)
- also here there is a `diff` test showing the results